### PR TITLE
Update ros.py to support CHR

### PR DESCRIPTION
**Pull request by nunomcruz**
Previous implementation supposed there was always a routerboard method, however this isn't true for CHR, where a serial number isn't also available.

### CODE REVIEW

**Comment by luqasz** [comment]
Hi. 

Sorry for my delayed reply. I will look at this closer in few days.

File: napalm_ros/ros.py [hunk 1]
```diff
@@ -354,12 +359,27 @@ def get_facts(self):
...
             'os_version': resource['version'],
-            'serial_number': routerboard.get('serial-number', ''),
+            'serial_number': serial_number,
             'interface_list': napalm.base.utils.string_parsers.sorted_nicely(
                 tuple(iface['name'] for iface in interfaces),
             ),
         }
 
+    
+    def get_config(self,retrieve='all', full=False, sanitized=False):
```
Comment (luqasz) [inline napalm_ros/ros.py:369]: Please split this into separate PR

**Review by luqasz — CHANGES_REQUESTED** [review]

File: napalm_ros/ros.py (hunk 1 above)
Comment (nunomcruz) [reply napalm_ros/ros.py]: Just reverted it, check if it's ok.

File: napalm_ros/ros.py [hunk 2]
```diff
@@ -345,7 +345,11 @@ def get_environment(self):
     def get_facts(self):
         resource = tuple(self.api('/system/resource/print'))[0]
         identity = tuple(self.api('/system/identity/print'))[0]
-        routerboard = tuple(self.api('/system/routerboard/print'))[0]
```
Comment (luqasz) [inline napalm_ros/ros.py:348]: Which exception is raised ? 
I'd prefer to use try, except here.

```python
try:
    routerboard = tuple(self.api('/system/routerboard/print'))[0]
except EXC:
    routerboard = dict()
```

**Review by luqasz — CHANGES_REQUESTED** [review]

File: napalm_ros/ros.py [hunk 3]
```diff
@@ -345,7 +345,11 @@ def get_environment(self):
...
+        if  resource['board-name'] == 'CHR':
```
Comment (luqasz) [inline napalm_ros/ros.py:348]: Fix formatting.

File: napalm_ros/ros.py (hunk 2 above)
Comment (nunomcruz) [reply napalm_ros/ros.py:348]: I agree with you but I was following the same style as other functions where self.api is called. I think that in this case get_facts should follow the same style as other functions, don't you agree? The same happens with other drivers (at least on srlinux and vyos), no exception handling is done during command calls. It's only performed during the connect phase.

**Comment by luqasz** [comment]
@nunomcruz Please rebase. I'fe fixed get_interfaces() error.

**Comment by luqasz** [comment]
Please update to newest master. I've fixed pylint errors.

**Pull request closed without merging**